### ISSUE TITLE
Why not set the output to be 1*128*128*128 

### ISSUE THREAD

**Issue by yianzhongguo** [ellisdg/3DUnetCNN#146]
Hi, sir. I appreciate your contribution and I have learned a lot from this program. But when I read the code I found the outputs were set to be n_labels*128*128*128. My question is why not set the output to be 1*128*128*128 since the brats data only contain one "truth" MRI. Thank you very much!

**Comment by athon2** [comment]
As far as I understand it, this is due to the use of dice loss, which turns the multi-label segmentation into three binary segmentation. The three 128*128*128 cubes output each correspond to a label(WT,ET,TC).

**Comment by yianzhongguo** [comment]
@athon2 Thank you. I agree with you. Thus I want to find  out how one "truth" MRI is split into three binary parts. But the division procession seems unconspicuous and is hidden in the original codes of Keras. I also looked into the original code of Keras and however it is too hard for me.

**Comment by athon2** [comment]
@yianzhongguo https://github.com/ellisdg/3DUnetCNN/blob/f913e27ec18b2a84298f49e88649486a60137056/unet3d/generator.py#L252  Maybe this is what you're looking for.

**Comment by yianzhongguo** [comment]
@athon2 Thank you very much! get_multi_class_labels(data, n_labels, labels=None) is the very function I am looking for. I have made a detour to check the source code of keras.

**Comment by yianzhongguo** [comment]
@athon2 Hi, athon2! I have another question: when I use the softmax function to replace the sigmoid function at the output layer as described in the paper "Brain Tumor Segmentation and Radiomics Survival Prediction: Contribution to the BRATS2017 Challenge", the loss of network will keep around -0.04 and will not decend at all. The question has been bothering me. Could you help me find the reasons, thank you!
